python/torch_mlir/__init__.py

```diff
@@ -141,6 +141,7 @@ def compile(model: torch.nn.Module,
             use_tracing: bool = False,
             ignore_traced_shapes = False,
             backend_legal_ops: Optional[Sequence[str]] = None,
+            backend_custom_ops: Optional[Sequence[str]] = None,
```

Comment:
@silvasean - I understand that `BACKEND_LEGAL_OPS` construct above is used to identify which operations are legal for each of the backends, but it does not distinguish between the operations that should be converted into custom ops and those that only the decomposition step will be skipped.

One option I had was to extend the `BACKEND_LEGAL_OPS` to have two lists for each backend: "decompose" and "custom". Where for the ops in the "decompose" list only the decomposition step will be skipped and for those in "custom", on top of skipping the decomposition, the backend will convert them to CustomOps.

Another option was using another list (like this patch). I am not sure if this aligns with your original idea regarding marking backend-legal and backend-custom operations.

Reply:
This is getting into power user territory. I don't think we should directly support this in `torch_mlir.compile`. We can add a `torch_mlir.tosa` module that builds this functionality on top of the base torch_mlir.compile API though (we could call the function `torch_mlir.tosa.compile`). It would mostly be a passthrough to torch_mlir.compile but would need to set the backend_legal_ops correctly and use `torch_mlir.compile` to compile to the backend contract, and then appropriately invoke the torch-backend-to-tosa-backend-pipeline.

Reply:
Done, added a module under `backends.tosa` and have it call `torch_mlir.compile` API. The `backend_custom_ops` argument is only present in the `torch_mlir.backends.tosa.compile` call.